### ISSUE TITLE
Error: Building package glibmm:x86-windows failed with: BUILD_FAILED

### ISSUE THREAD

**Issue by t-sharp** [Microsoft/vcpkg#2249]
I performed ".\vcpkg update" and then ".\vcpkg install glibmm". The output was:
>The following packages will be built and installed:
>    glibmm:x86-windows
>Starting package 1/1: glibmm:x86-windows
>Building package glibmm:x86-windows...
>-- CURRENT_INSTALLED_DIR=C:/Program Files (x86)/vcpkg/vcpkg/installed/x86-windows
>-- DOWNLOADS=C:/Program Files (x86)/vcpkg/vcpkg/downloads
>-- CURRENT_PACKAGES_DIR=C:/Program Files (x86)/vcpkg/vcpkg/packages/glibmm_x86-windows
>-- CURRENT_BUILDTREES_DIR=C:/Program Files (x86)/vcpkg/vcpkg/buildtrees/glibmm
>-- CURRENT_PORT_DIR=C:/Program Files (x86)/vcpkg/vcpkg/ports/glibmm/.
>-- Using cached C:/Program Files (x86)/vcpkg/vcpkg/downloads/glibmm-2.52.1.tar.xz
>-- Testing integrity of cached file...
>-- Testing integrity of cached file... OK
>-- Extracting done
>-- Applying patch C:/Program Files (x86)/vcpkg/vcpkg/ports/glibmm/fix_properties.patch
>-- Applying patch failed. This is expected if this patch was previously applied.
>-- Applying patch C:/Program Files (x86)/vcpkg/vcpkg/ports/glibmm/fix_properties.patch done
>-- Applying patch C:/Program Files (x86)/vcpkg/vcpkg/ports/glibmm/fix_charset.patch
>-- Applying patch failed. This is expected if this patch was previously applied.
>-- Applying patch C:/Program Files (x86)/vcpkg/vcpkg/ports/glibmm/fix_charset.patch done
>-- Building C:/Program Files (x86)/vcpkg/vcpkg/buildtrees/glibmm/src/glibmm-2.52.1/MSVC_Net2013/glibmm.sln for Release
>CMake Error at C:/Program Files (x86)/vcpkg/vcpkg/scripts/cmake/vcpkg_execute_required_process.cmake:43 (message):
>    Command failed: msbuild;C:/Program Files (x86)/vcpkg/vcpkg/buildtrees/glibmm/src/glibmm-2.52.1/MSVC_Net2013/glibmm.sln;/p:Configuration=Release;/p:ForceImportBeforeCppTargets=C:/Program Files (x86)/vcpkg/vcpkg/scripts/buildsystems/msbuild/vcpkg.targets;/t:giomm;/p:Platform=Win32;/p:PlatformToolset=v141;/p:VCPkgLocalAppDataDisabled=true;/p:UseIntelMKL=No;/p:WindowsTargetPlatformVersion=10.0.16299.0;/m
>    Working Directory: C:/Program Files (x86)/vcpkg/vcpkg/buildtrees/glibmm/x86-windows-rel
>    See logs for more information:
>      C:\Program Files (x86)\vcpkg\vcpkg\buildtrees\glibmm\build-x86-windows-rel-out.log
>      C:\Program Files (x86)\vcpkg\vcpkg\buildtrees\glibmm\build-x86-windows-rel-err.log
>
>Call Stack (most recent call first):
>  C:/Program Files (x86)/vcpkg/vcpkg/scripts/cmake/vcpkg_build_msbuild.cmake:92 (vcpkg_execute_required_process)
>  C:/Program Files (x86)/vcpkg/vcpkg/ports/glibmm/portfile.cmake:33 (vcpkg_build_msbuild)
>  C:/Program Files (x86)/vcpkg/vcpkg/scripts/ports.cmake:72 (include)
>
>
>Error: Building package glibmm:x86-windows failed with: BUILD_FAILED
>Please ensure you're using the latest portfiles with \`.\vcpkg update\`, then
>submit an issue at https://github.com/Microsoft/vcpkg/issues including:
>  Package: glibmm:x86-windows
>  Vcpkg version: 0.0.97-2017-11-17-06819e43e30b64769d1ef036d20f6038dab4da34
>
>Additionally, attach any relevant sections from the log files above.

But I'm unable to attach the logs to this issue. When I try to attach "build-x86-windows-rel-err.log" GitHub says that this file is empty. The other file is "build-x86-windows-rel-out.log" but GitHub says "Something went really wrong, and we can't process that file. Try again" and every time I try again it doesn't work (even if I change the file extension to ".txt").

If you want the file, you can ask me to quote it.
Insted I just quoted you the first message that sounds strange:
>Description: Generate glibmm def file
>         C:\Program Files (x86)\vcpkg\vcpkg\buildtrees\glibmm\src\glibmm-2.52.1\MSVC_Net2013\Release\Win32\bin\\gendef.exe C:\Program Files (x86)\vcpkg\vcpkg\buildtrees\glibmm\src\glibmm-2.52.1\MSVC_Net2013\Release\Win32\obj\glibmm\\glibmm.def glibmm.dll C:\Program Files (x86)\vcpkg\vcpkg\buildtrees\glibmm\src\glibmm-2.52.1\MSVC_Net2013\Release\Win32\obj\glibmm\*.obj
>         :VCEnd
>         Der Befehl "C:\Program" ist entweder falsch geschrieben oder
>         konnte nicht gefunden werden.

Sorry for writing you a German sentence, but I'm using a German version of Windows 10.
The last sentence translated into English would be similar to "The command "C:\Program" is either misspelled or could not be found." 
Sorry for my bad English, but I think that this issue happend because my path of vcpkg contains whitespaces. Because vcpkg is located at "C:\Program Files (x86)\vcpkg\vcpkg\" on my PC.

Let me know if there's any more information I can provide.

**Comment by tpdussla93** [comment]
I have a same Error installing package curl.
The output was:

C:\vcpkg-master>.\vcpkg install curl
The following packages will be built and installed:
    curl:x86-windows
  * libssh2:x86-windows
  * openssl:x86-windows
  * zlib:x86-windows
Additional packages (*) will be installed to complete this operation.
Starting package 1/4: zlib:x86-windows
Building package zlib:x86-windows...
A suitable version of cmake was not found (required v3.10.0). Downloading portable cmake v3.10.0...
A suitable version of git was not found (required v2.15.0). Downloading portable git v2.15.0...
-- CURRENT_INSTALLED_DIR=C:/vcpkg-master/installed/x86-windows
-- DOWNLOADS=C:/vcpkg-master/downloads
-- CURRENT_PACKAGES_DIR=C:/vcpkg-master/packages/zlib_x86-windows
-- CURRENT_BUILDTREES_DIR=C:/vcpkg-master/buildtrees/zlib
-- CURRENT_PORT_DIR=C:/vcpkg-master/ports/zlib/.
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz...
-- Downloading http://www.zlib.net/zlib-1.2.11.tar.gz... OK
-- Testing integrity of downloaded file...
-- Testing integrity of downloaded file... OK
-- Extracting source C:/vcpkg-master/downloads/zlib1211.tar.gz
-- Extracting done
-- Applying patch C:/vcpkg-master/ports/zlib/cmake_dont_build_more_than_needed.patch
-- Applying patch C:/vcpkg-master/ports/zlib/cmake_dont_build_more_than_needed.patch done
-- Downloading https://github.com/ninja-build/ninja/releases/download/v1.8.2/ninja-win.zip...
-- Downloading https://github.com/ninja-build/ninja/releases/download/v1.8.2/ninja-win.zip... OK
-- Testing integrity of downloaded file...
-- Testing integrity of downloaded file... OK
-- Configuring x86-windows-rel
-- Configuring x86-windows-rel done
-- Configuring x86-windows-dbg
-- Configuring x86-windows-dbg done
-- Build x86-windows-rel
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:44 (message):
    Command failed: C:/vcpkg-master/downloads/cmake-3.10.0-win32-x86/bin/cmake.exe;--build;.;--config;Release;--target;install;--;-v
    Working Directory: C:/vcpkg-master/buildtrees/zlib/x86-windows-rel
    See logs for more information:
      C:\vcpkg-master\buildtrees\zlib\install-x86-windows-rel-out.log
      C:\vcpkg-master\buildtrees\zlib\install-x86-windows-rel-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:63 (vcpkg_execute_required_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/zlib/portfile.cmake:26 (vcpkg_install_cmake)
  scripts/ports.cmake:72 (include)


Error: Building package zlib:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: zlib:x86-windows
  Vcpkg version: 0.0.99-unknownhash

Additionally, attach any relevant sections from the log files above.

Then I attach the log files.
[install-x86-windows-rel-out.log](https://github.com/Microsoft/vcpkg/files/1514311/install-x86-windows-rel-out.log)

**Comment by tpdussla93** [comment]
I have resolved this problem.
After updating by Visual Studio Installer, that error had not occurred.

**Comment by fishfreetaken** [comment]
@tpdussla93 Can you explain for me how to update by Visual Studio Installer.
I fall in the same problem as you for curl installation!

**Comment by tpdussla93** [comment]
@fishfreetaken 
I updated Visual Studio by Visual Studio Installer at start menu.
![image](https://user-images.githubusercontent.com/32669846/33822347-8d169140-de9a-11e7-8045-b81ca29e486d.png)
If you launch the Visual Studio Installer you can find update button.
Then maybe you should install "Linux developement using C++".

![image](https://user-images.githubusercontent.com/32669846/33824916-3e342d90-dea3-11e7-9a1e-4c37e870c806.png)

**Comment by t-sharp** [comment]
Hello @tpdussla93 
sorry for my really late reply but I'm a student and therefore don't have much time.
I followed your steps and updated both Visual Studio Installer and Visual Studio. I updated vcpkg again and the result is the same: I'm still unable to build package "glibmm"
And (unfortunately) I'm still unable to attach the log files. GitHub still says "Something went really wrong, and we can't process that file."
I think that the problem is still the same: The path of my vcpkg-installation contains whitespaces, because I tried a second installation of vcpkg on my external hard drive in a path without any whitespaces and it worked.

I'm closing this issue now, because I don't think this issue will ever be solved.
But thanks for your comments and I wish you a nice christmas eve.
